addons/im_livechat/controllers/rtc.py

```diff
@@ -0,0 +1,19 @@
+from werkzeug.exceptions import NotFound
+
+from odoo.http import route, request
+from odoo.addons.mail.controllers.discuss.rtc import RtcController
+from odoo.addons.mail.models.discuss.mail_guest import add_guest_to_context
+
+
+class LivechatRtcController(RtcController):
+    @route()
+    @add_guest_to_context
+    def channel_call_join(self, channel_id, check_rtc_session_ids=None, camera=False):
+        channel = request.env["discuss.channel"].search([
+            ("id", "=", channel_id),
+            ("channel_type", "=", "livechat"),
+            ("rtc_session_ids", "=", False),
+        ])
+        if channel and request.env.user.is_public:
+            raise NotFound()
```

Comment:
This query is useless if the user is not public, we should be checked that before hand.